From 2345b020b862ec855038dd32a51ebb072647f28d Mon Sep 17 00:00:00 2001
From: fuscated <fuscated@2a5c6006-c6dd-42ca-98ab-0921f2732cef>
Date: Sun, 22 Mar 2020 20:02:49 +0000
Subject: [PATCH] - build: Fixes to make it compile when using wx-master

> Paint events aren't no longer allowed to be created. We've used this in a
  single place in the code. I'm not sure what is the idea behind this,
  probably to redraw something. For now I'm removing it and we'll see if
  something breaks.
> The second change is the removal of a c-tor in wxPGWindowList, I've
  replaced it with the non-generic two parameter one.

git-svn-id: https://svn.code.sf.net/p/codeblocks/code/trunk@11991 2a5c6006-c6dd-42ca-98ab-0921f2732cef
---
 src/src/main.cpp       | 2 --
 src/src/watchesdlg.cpp | 3 +--
 2 files changed, 1 insertion(+), 4 deletions(-)

diff --git a/src/src/main.cpp b/src/src/main.cpp
index 7d0b317eb..323457761 100644
--- a/src/src/main.cpp
+++ b/src/src/main.cpp
@@ -2680,8 +2680,6 @@ bool MainFrame::OnDropFiles(wxCoord /*x*/, wxCoord /*y*/, const wxArrayString& f
     else
     {
         wxBusyCursor useless;
-        wxPaintEvent e;
-        ProcessEvent(e);
         for (unsigned int i = 0; i < files.GetCount(); ++i)
           success &= OpenGeneric(files[i]);
     }
diff --git a/src/src/watchesdlg.cpp b/src/src/watchesdlg.cpp
index 9ff3e8655..ca4fd979a 100644
--- a/src/src/watchesdlg.cpp
+++ b/src/src/watchesdlg.cpp
@@ -103,8 +103,7 @@ class cbDummyEditor : public wxPGEditor
                                   cb_unused const wxPoint& pos,
                                   cb_unused const wxSize& sz) const override
     {
-        wxPGWindowList const list;
-        return list;
+        return wxPGWindowList(nullptr, nullptr);
     }
     void UpdateControl(cb_unused wxPGProperty* property, cb_unused wxWindow* ctrl) const override {}
     bool OnEvent(cb_unused wxPropertyGrid* propgrid, cb_unused wxPGProperty* property,
